pbm/storage/s3/s3.go

```diff
@@ -0,0 +1,237 @@
+package s3
+
+import (
+	"io"
+	"io/ioutil"
+	"net/url"
+	"path"
+	"runtime"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/minio/minio-go"
+	"github.com/pkg/errors"
+)
+
+const (
+	// GCSEndpointURL is the endpoint url for Google Clound Strage service
+	GCSEndpointURL = "storage.googleapis.com"
+
+	defaultS3Region = "us-east-1"
+)
+
+type Conf struct {
+	Provider    S3Provider  `bson:"provider,omitempty" json:"provider,omitempty" yaml:"provider,omitempty"`
+	Region      string      `bson:"region" json:"region" yaml:"region"`
+	EndpointURL string      `bson:"endpointUrl,omitempty" json:"endpointUrl" yaml:"endpointUrl,omitempty"`
+	Bucket      string      `bson:"bucket" json:"bucket" yaml:"bucket"`
+	Prefix      string      `bson:"prefix,omitempty" json:"prefix,omitempty" yaml:"prefix,omitempty"`
+	Credentials Credentials `bson:"credentials" json:"credentials,omitempty" yaml:"credentials"`
+}
+
+func (c *Conf) Cast() error {
```

Comment:
Not sure I understood it right, but looks like this func is only aboul S3 config configuration/validation. If so why not put it in Conf `New` func and do unmarshaling there. Because now this func looks like secret knowledge.